### PR TITLE
Murisi/shared masp2

### PR DESCRIPTION
**Pull request by murisi**
Factored the platform specific components out of `ShieldedContext` and then moved `ShieldedContext` into the `shared` crate. Essentially addressing: https://github.com/anoma/namada/issues/302 .

### CODE REVIEW

File: shared/src/ledger/masp.rs [hunk 1]
```diff
@@ -200,3 +246,1044 @@ pub fn get_params_dir() -> PathBuf {
...
+    async fn query_storage_value<T: Send>(
+        &self,
+        key: &storage::Key,
+    ) -> Option<T>
+    where
+        T: BorshDeserialize;
+
+    async fn query_epoch(&self) -> Epoch;
+
+    fn local_tx_prover(&self) -> LocalTxProver;
```
Comment (memasdeligeorgakis) [inline shared/src/ledger/masp.rs:261]: Is this specific to the consumer? Something that would differ from what is being done on desktop/CLI?

File: shared/src/ledger/masp.rs [hunk 2]
```diff
@@ -200,3 +246,1044 @@ pub fn get_params_dir() -> PathBuf {
...
+
+    async fn query_conversion(
+        &self,
+        asset_type: AssetType,
+    ) -> Option<(
+        Address,
+        Epoch,
+        masp_primitives::transaction::components::Amount,
+        MerklePath<Node>,
+    )>;
```
Comment (memasdeligeorgakis) [inline shared/src/ledger/masp.rs:275]: Also here, does this mean that we should define this in a web/WASM specific way?

```rust
    async fn query_conversion(
        &self,
        asset_type: AssetType,
    ) -> Option<(
        Address,
        Epoch,
        masp_primitives::transaction::components::Amount,
        MerklePath<Node>,
    )>;
```

**Comment by cwgoes** [comment]
@murisi @mariari is this PR still applicable?

**Comment by mariari** [comment]
> @murisi @mariari is this PR still applicable?

The work in #925 (integration branch) / #921 (branch proper)  follows this work up given the commit log @murisi would know more about the work committed after I went on break.

**Comment by murisi** [comment]
> @murisi @mariari is this PR still applicable?

@mariari is correct. The initial scope of my work was to put the MASP transaction code into an SDK. After completing that, the scope then broadened into putting CLI subcommands into an SDK.  Hence this branch is followed up/superseded by #925 and #921 .

**Pull request closed without merging**